src/platform_impl/linux/x11/util/input.rs

```diff
@@ -17,12 +17,12 @@ impl ModifiersState {
     }
 
     pub(crate) fn from_x11_mask(mask: c_uint) -> Self {
-        ModifiersState {
-            alt: mask & ffi::Mod1Mask != 0,
-            shift: mask & ffi::ShiftMask != 0,
-            ctrl: mask & ffi::ControlMask != 0,
-            logo: mask & ffi::Mod4Mask != 0,
-        }
+        let mut m = ModifiersState::empty();
```

Comment:
And how would you distinguish between the left and right keys on x11?

Reply:
X11 can check against hard-coded keycode values for left and right Ctrl, Shift, etc. However, it would be helpful if the bit mask had three bits for each modifier; e.g. `0b001` => ctrl modifier set, `0b010` => left ctrl, `0b100` => right ctrl. This way, X11 code could express that a modifier is set even when the modifier key does not correspond to a known keycode. Documentation would also need to reflect that `mods.ctrl()` may be `true` when both `mods.left_ctrl()` and `mods.right_ctrl()` are `false`.

Reply:
I've updated the patterns to use three bits. This PR should be ready to review again.